modules/promotion/src/Entity/Promotion.php

```diff
@@ -176,6 +176,16 @@ public function getOffer() {
   /**
    * {@inheritdoc}
    */
+  public function setOffer($target_plugin_id, array $target_plugin_configuration) {
```

Comment:
Thanks for the suggestion.

```php
// Returns array with plugin id and target config value.
$offer_plugin_config_array =  $promotion->get('offer')->first()->getValue();

// Returns offer plugin object of type PromotionOfferBase. Can we get all info from this object???
$offer_plugin_object =  $promotion->getOffer();
```

@mglaman which one should be passed to setOffer method?? as per my knowledge, $offer_object can be enforced for typehint but not sure if we get all reqd. info. from this object. i'm talking about configuration like amount etc. getConfiguration() doesn't help here, gives different keys.

if we pass the $offer_plugin_config_array, i'm not sure how we can enforce if the reqd. values are present and if user knows to get this value from??